app/forms/punches_create_form.rb

```diff
@@ -0,0 +1,46 @@
+class PunchesCreateForm
+  include ActiveModel::Model
+
+  attr_accessor :punches
+
+  validates :punches, presence: true
+  validate :punches_validations, :punch_on_same_day_validation
+
+  def initialize(params)
+    params ||= {}
+    params[:punches] ||= []
+    params[:user] ||= User.new
```

Comment:
```suggestion
  def initialize(punches: [], user: User.new)
```
You can use the keyword syntax to reduce some code.

This will still expect a hash to be passed, but your method already has access to the variables directly.

Reply:
Thanks, I didnt know about that